### PR TITLE
add set_iterate_upper_bound api for readoptions

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
"iterate_upper_bound" defines the extent up to which the forward iterator can returns entries. Once the bound is reached, Valid() will be false.

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
@BusyJay @siddontang @ngaut PTAL

**Comment by siddontang** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
@siddontang @BusyJay @ngaut PTAL

File: src/rocksdb.rs [hunk 1]
```diff
@@ -1081,6 +1085,15 @@ impl ReadOptions {
         rocksdb_ffi::rocksdb_readoptions_set_snapshot(self.inner,
                                                       snapshot.inner);
     }
+
+    pub fn set_iterate_upper_bound(&mut self, key: &str) {
```
Comment (BusyJay) [inline src/rocksdb.rs]: Use `&[u8]` instead.

**Comment by BusyJay** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**